api/common/sharing/utils.go

```diff
@@ -135,11 +132,47 @@ func getSharedInstances(ctx context.Context, repository storage.Repository, offe
 	return objectList, err
 }
 
-func filterInstancesBySelectors(ctx context.Context, repository storage.Repository, instances types.ObjectList, parameters map[string]gjson.Result, smaap bool, offeringID string) (*types.ServiceInstance, error) {
+func filterInstancesBySelectors(ctx context.Context, repository storage.Repository, offeringID, tenantIdentifier, tenantValue string, selectors map[string]gjson.Result, smaap bool) (*types.ServiceInstance, error) {
 	var err error
 	var filteredInstances types.ServiceInstances
 
-	planSelectorVal, planSelectorExists := parameters[instance_sharing.ReferencePlanNameSelectorKey]
+	var sharedInstances types.ObjectList
+
+	// Filter by label query:
+	labelsSelector, labelExists := selectors[instance_sharing.ReferenceLabelSelectorKey]
+	labelsSelectorArray := labelsSelector.Array()
+	hasLabelsSelectors := labelExists && len(labelsSelectorArray) > 0 && labelsSelectorArray[0].String() != ""
+	if hasLabelsSelectors {
+		var criteria []query.Criterion
```

Comment:
criterias

Reply:
criteria are already plural
criterion is it's singular